app/models/intake.rb

```diff
@@ -451,12 +451,7 @@ def requested_docs_token_link
   end
 
   def filing_years
-    [
-      ("2019" if needs_help_2019_yes?),
-      ("2018" if needs_help_2018_yes?),
-      ("2017" if needs_help_2017_yes?),
-      ("2016" if needs_help_2016_yes?),
-    ].compact
+    TaxReturn.filing_years.map { |year| year if send("needs_help_#{year}_yes?") }.compact.map(&:to_s)
```

Comment:
I like how there is now an authoritative, centralized list of tax years. While the previous code was in my opinion a little more readable, I think the new version is more future-proof and maintainable.